### ISSUE TITLE
Encountered a Dart Analysis Server error.

### ISSUE THREAD

**Issue by hdaya** [dart-lang/sdk#38327]
Analyzer Feedback from IntelliJ

## Version information

- `IDEA AI-191.8026.42.35.5791312`
- `2.5.0-edge.1ef83b86ae637ffe7359173804cbc6d3fa25e6db`
- `AI-191.8026.42.35.5791312, JRE 1.8.0_202-release-1483-b03x64 JetBrains s.r.o, OS Windows 10(amd64) v10.0 , screens 1366x768`

## Exception
```
Dart analysis server, SDK version 2.5.0-edge.1ef83b86ae637ffe7359173804cbc6d3fa25e6db, server version 1.27.2, error: Captured exception
type 'MethodInvocationImpl' is not a subtype of type 'ThisExpression' in type cast
#0      AstBuilder.buildInitializer (package:analyzer/src/fasta/ast_builder.dart:348:23)
#1      AstBuilder.endInitializers (package:analyzer/src/fasta/ast_builder.dart:1369:44)
#2      Parser.parseInitializers (package:front_end/src/fasta/parser/parser.dart:2586:14)
#3      Parser.parseInitializersOpt (package:front_end/src/fasta/parser/parser.dart:2523:14)
#4      Parser.parseMethod (package:front_end/src/fasta/parser/parser.dart:3250:13)
#5      Parser.parseClassOrMixinMemberImpl (package:front_end/src/fasta/parser/parser.dart:3130:15)
#6      Parser.parseClassOrMixinBody (package:front_end/src/fasta/parser/parser.dart:2906:15)
#7      Parser.parseClass (package:front_end/src/fasta/parser/parser.dart:1797:13)
#8      Parser.parseClassOrNamedMixinApplication (package:front_end/src/fasta/parser/parser.dart:1757:14)
#9      Parser.parseTopLevelKeywordDeclaration (package:front_end/src/fasta/parser/parser.dart:572:14)
#10     Parser.parseTopLevelDeclarationImpl (package:front_end/src/fasta/parser/parser.dart:468:14)
#11     Parser.parseUnit (package:front_end/src/fasta/parser/parser.dart:350:15)
#12     ParserAdapter.parseCompilationUnit2 (package:analyzer/src/generated/parser_fasta.dart:157:32)
#13     ParserAdapter.parseCompilationUnit (package:analyzer/src/generated/parser_fasta.dart:152:12)
#14     _File._parse (package:analyzer/src/services/available_declarations.dart:1839:23)
#15     _File.refresh (package:analyzer/src/services/available_declarations.dart:1160:30)
#16     DeclarationsTracker._getFileByPath (package:analyzer/src/services/available_declarations.dart:660:14)
#17     DeclarationsTracker.doWork (package:analyzer/src/services/available_declarations.dart:573:18)
#18     CompletionLibrariesWorker.performWork (package:analysis_server/src/domains/completion/available_suggestions.dart:283:13)
<asynchronous suspension>
#19     AnalysisDriverScheduler._run (package:analyzer/src/dart/analysis/driver.dart:1901:35)
<asynchronous suspension>
#20     AnalysisDriverScheduler.start (package:analyzer/src/dart/analysis/driver.dart:1855:5)
#21     new AnalysisServer (package:analysis_server/src/analysis_server.dart:212:29)
#22     SocketServer.createAnalysisServer (package:analysis_server/src/socket_server.dart:86:26)
#23     StdioAnalysisServer.serveStdio (package:analysis_server/src/server/stdio_server.dart:37:18)
#24     Driver.startAnalysisServer.<anonymous closure> (package:analysis_server/src/server/driver.dart:572:21)
#25     _rootRun (dart:async/zone.dart:1124:13)
#26     _CustomZone.run (dart:async/zone.dart:1021:19)
#27     _runZoned (dart:async/zone.dart:1516:10)
#28     runZoned (dart:async/zone.dart:1463:12)
#29     Driver._captureExceptions (package:analysis_server/src/server/driver.dart:689:12)
#30     Driver.startAnalysisServer (package:analysis_server/src/server/driver.dart:570:7)
#31     Driver.start.<anonymous closure> (package:analysis_server/src/server/driver.dart:474:9)
#32     _AsyncAwaitCompleter.start (dart:async-patch/async_patch.dart:43:6)
#33     Driver.start.<anonymous closure> (package:analysis_server/src/server/driver.dart:469:43)
#34     CompilerContext.runInContext.<anonymous closure>.<anonymous closure> (package:front_end/src/fasta/compiler_context.dart:122:46)
#35     new Future.sync (dart:async/future.dart:224:31)
#36     CompilerContext.runInContext.<anonymous closure> (package:front_end/src/fasta/compiler_context.dart:122:19)
#37     _rootRun (dart:async/zone.dart:1124:13)
#38     _CustomZone.run (dart:async/zone.dart:1021:19)
#39     _runZoned (dart:async/zone.dart:1516:10)
#40     runZoned (dart:async/zone.dart:1463:12)
#41     CompilerContext.runInContext (package:front_end/src/fasta/compiler_context.dart:121:12)
#42     CompilerContext.runWithDefaultOptions (package:front_end/src/fasta/compiler_context.dart:140:56)
#43     Driver.start (package:analysis_server/src/server/driver.dart:469:21)
#44     main (file:///C:/b/s/w/ir/k/src/third_party/dart/pkg/analysis_server/bin/server.dart:12:11)
#45     _AsyncAwaitCompleter.start (dart:async-patch/async_patch.dart:43:6)
#46     main (file:///C:/b/s/w/ir/k/src/third_party/dart/pkg/analysis_server/bin/server.dart:10:10)
#47     _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:303:32)
#48     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:172:12)
```

For additional log information, please append the contents of
file://C:\Users\makkawi\AppData\Local\Temp\report4617.txt.

**Comment by keertip** [comment]
@scheglov, this is from a bleeding edge build 19 days ago. Maybe this has been fixed?

**Comment by scheglov** [comment]
It looks like it might have been fixed in bfc7f8c4b569cfa612c33a059414e8e1cfc86c95.
